File: app/src/api/env_group.ts
import type { ModelBase, UpdateOrderRequest } from '@/api/curd'
import { extendCurdApi, http, useCurdApi } from '@uozi-admin/request'

// Post-sync action types
export const PostSyncAction = {
  None: 'none',
  ReloadNginx: 'reload_nginx',
}

// Upstream test types
export const UpstreamTestType = {
  Local: 'local',
  Remote: 'remote',
  Mirror: 'mirror',
}

export interface EnvGroup extends ModelBase {
  name: string
  sync_node_ids: number[]
  post_sync_action?: string
  upstream_test_type?: string
}

const baseUrl = '/env_groups'

const env_group = extendCurdApi(useCurdApi<EnvGroup>(baseUrl), {
  updateOrder(data: UpdateOrderRequest) {
    return http.post('/env_groups/order', data)
  },
})

export default env_group

File: model/env_group.go
package model

// PostSyncActionType defines the type of action after synchronization
const (
	// PostSyncActionNone indicates no operation after sync
	PostSyncActionNone = "none"
	// PostSyncActionReloadNginx indicates reload Nginx after sync
	PostSyncActionReloadNginx = "reload_nginx"
)

// UpstreamTestType defines the type of upstream test
const (
	// UpstreamTestLocal indicates local upstream test
	UpstreamTestLocal = "local"
	// UpstreamTestRemote indicates remote upstream test
	UpstreamTestRemote = "remote"
	// UpstreamTestMirror indicates mirror upstream test
	UpstreamTestMirror = "mirror"
)

// EnvGroup represents a group of environments that can be synced across nodes
type EnvGroup struct {
	Model
	Name             string   `json:"name"`
	SyncNodeIds      []uint64 `json:"sync_node_ids" gorm:"serializer:json"`
	OrderID          int      `json:"-" gorm:"default:0"`
	PostSyncAction   string   `json:"post_sync_action" gorm:"default:'reload_nginx'"`
	UpstreamTestType string   `json:"upstream_test_type" gorm:"default:'local'"`
}
